ethereum/permissioning/src/main/java/tech/pegasys/pantheon/ethereum/permissioning/NodePermissioningControllerFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.permissioning;
+
+import tech.pegasys.pantheon.ethereum.core.Synchronizer;
+import tech.pegasys.pantheon.ethereum.permissioning.node.NodePermissioningController;
+import tech.pegasys.pantheon.ethereum.permissioning.node.NodePermissioningProvider;
+import tech.pegasys.pantheon.ethereum.permissioning.node.provider.SyncStatusNodePermissioningProvider;
+import tech.pegasys.pantheon.ethereum.transaction.TransactionSimulator;
+import tech.pegasys.pantheon.util.enode.EnodeURL;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class NodePermissioningControllerFactory {
+
+  public NodePermissioningController create(
+      final PermissioningConfiguration permissioningConfiguration,
+      final Synchronizer synchronizer,
+      final Collection<EnodeURL> bootnodes,
+      final TransactionSimulator transactionSimulator) {
+    final SyncStatusNodePermissioningProvider syncStatusProvider =
```

Comment:
We should only use `SyncStatusNodePermissioningProvider` if the smart contract based permissioning is enabled.

Reply:
done